src/rect.js

```diff
@@ -0,0 +1,33 @@
+import React, { memo } from 'react'
+import { useThemeUI } from 'theme-ui'
+import { useChart } from './chart'
+
+const Rect = ({ x, y, color = 'primary', ...props }) => {
+  const { x: _x, y: _y } = useChart()
+  const { theme } = useThemeUI()
+
+  if (x[0] > x[1]) {
+    x = x.reverse()
+  }
+
+  if (y[0] > y[1]) {
+    y = y.reverse()
+  }
+
+  const width = Math.abs(x[1] - x[0])
+  const height = Math.abs(y[1] - y[0])
+
+  const h = _x(x[0] + width) - _x(x[0])
+  const v = _y(y[0]) - _y(y[0] + height)
+  return (
+    <path
+      key={i}
+      {...props}
```

Comment:
See my comment on `Bar` re: the order here. Seems especially useful for `Rect` to have the order changed in order to directly specify, say, the stroke. Or maybe I'm misinterpreting the consequence of the ordering!